src/main/java/org/codeforamerica/messaging/services/MessageService.java

```diff
@@ -108,21 +95,33 @@ public Message saveMessage(MessageRequest messageRequest, MessageBatch messageBa
 
     public void sendMessage(Long messageId) {
         log.info("Sending message #{}", messageId);
-        try {
-            Message message = messageRepository.findById(messageId).get();
-            if (message.needToSendSms()) {
-                SmsMessage sentSmsMessage = this.smsService.sendSmsMessage(message.getToPhone(), message.getSmsBody());
+        Message message = messageRepository.findById(messageId).get();
+        TemplateVariant templateVariant = message.getTemplateVariant();
+        Map<String, String> templateParams = message.getTemplateParams();
+        if (message.needToSendSms()) {
+            try {
+                String smsBody = templateVariant.build(TemplateVariant::getSmsBody, templateParams);
+                SmsMessage sentSmsMessage = this.smsService.sendSmsMessage(message.getToPhone(), smsBody);
                 message.setSmsMessage(sentSmsMessage);
                 messageRepository.save(message);
+            } catch (IOException e) {
```

Comment:
Are we sure that `IOException` will not be thrown by other issues besides template processing? If we aren't, it would be better to just have the handler for `Exception`. If the stack trace is insufficient, we could add the templateVariant id to that log message.

Reply:
Since the template params are already parsed out I wouldn't think there would be any other IOExceptions, but the distinction may not be very useful so I collapsed them.